generators/app/prompts.js

```diff
@@ -62,16 +72,53 @@ async function askForApplicationType() {
     },
   ];
 
-  const answers = await this.prompt([
-    {
-      type: 'list',
-      name: 'applicationType',
-      message: `Which ${chalk.yellow('*type*')} of application would you like to create?`,
-      choices: applicationTypeChoices,
-      default: appDefaultConfig.applicationType,
-    },
-  ]);
-  this.applicationType = this.jhipsterConfig.applicationType = answers.applicationType;
+  await this.prompt(
+    [
+      {
+        type: 'list',
+        name: 'applicationType',
+        message: `Which ${chalk.yellow('*type*')} of application would you like to create?`,
+        choices: applicationTypeChoices,
+        default: appDefaultConfig.applicationType,
+      },
+      {
+        when: answers => {
+          const { applicationType } = answers;
+          const askForMicrofrontend = [GATEWAY, MICROSERVICE].includes(applicationType);
+          if (!askForMicrofrontend) {
+            answers.microfrontend = false;
+          }
+          return askForMicrofrontend;
+        },
+        type: 'confirm',
+        name: 'microfrontend',
+        message: `Do you want to enable ${chalk.yellow('microfrontends')}?`,
+        default: defaultConfigMicroservice.microfrontend,
+      },
+      {
+        when: answers => {
+          const { applicationType, microfrontend, microfrontends } = answers;
+          const askForMicrofrontends = applicationType === GATEWAY && microfrontend;
+          if (askForMicrofrontends) {
+            answers.microfrontends = microfrontendsToPromptValue(microfrontends);
+          } else {
+            answers.microfrontends = [];
+          }
+          return askForMicrofrontends;
+        },
+        type: 'input',
+        name: 'microfrontends',
+        message: `Comma separated ${chalk.yellow('microfrontends')} base names.`,
```

Comment:
As a developer, I don't really think of base names. I think of them more as microservice or app names.
<img width="1330" alt="Screen Shot 2022-06-07 at 22 24 57" src="https://user-images.githubusercontent.com/17892/172531588-9e221e3f-4775-44b2-b5ce-57b5ee51ff2c.png">
```suggestion
        message: `Comma separated ${chalk.yellow('*microfrontend*')} app names.`,
```